pkg/cloudproviders/k8s/k8s.go

```diff
@@ -0,0 +1,380 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/util/flowcontrol"
+
+	"github.com/ash2k/stager/wait"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/atlassian/gostatsd"
+	"github.com/atlassian/gostatsd/pkg/util"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	core_v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	core_v1inf "k8s.io/client-go/informers/core/v1"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+var (
+	// DefaultAnnotationTagWhitelist is the default ParamAnnotationTagWhitelist. Everything beginning with a standard
+	// gostatsd string is included by default.
+	DefaultAnnotationTagWhitelist = []string{
+		fmt.Sprintf("^%s", AnnotationPrefix),
+	}
+	// DefaultLabelTagWhitelist is the default ParamLabelTagWhitelist. Every label is ignored by default.
+	DefaultLabelTagWhitelist []string
+)
+
+const (
+	// ProviderName is the name of the k8s cloud provider.
+	ProviderName = "k8s"
+	// PodsByIPIndexName is the name of the index function storing pods by IP.
+	PodsByIPIndexName = "PodByIP"
+	// AnnotationPrefix is the annotation prefix that is turned into tags by default.
+	AnnotationPrefix = "gostatsd.atlassian.com/"
```

Comment:
Good catch! I think rather than making this constant a regex it should be escaped where it is used as part of a regex using https://golang.org/pkg/regexp/#QuoteMeta
WDYT?

Reply:
Great catch. I think keeping the constant a string for ease of reading and then escaping it on-use is the way. Thanks both of you!